### PR TITLE
Language support for LLVM Machine IR

### LINKED ISSUE #4762: [LANGUAGE REQUEST]: LLVM Machine IR
### Language name

LLVM Machine IR

### Language version

_No response_

### Language homepage

https://llvm.org/docs/MIRLangRef.html

### Compiler homepage

https://releases.llvm.org/

### Compiler version

LLVM 15.0.0

### Motivation

As a backend compiler engineer, I repeatedly run into the need for a compiler explorer for LLVM machine-level IR similar to what we have for LLVM IR. 
Since this MIR will need the same compiler that LLVM IR needs (opt & llc) it's straightforward to integrate. I have a working version for MIR support on a local instance and would like to submit a patch here.

### PR DESCRIPTION
**Pull request by yashssh**
Resolves #4762

### CODE REVIEW

**Review by jeremy-rifkin — APPROVED** [review]
Thank you so much! Tested locally and looks good to me

File: etc/config/llvm_mir.amazon.properties
Comment (RubenRBS) [inline etc/config/llvm_mir.amazon.properties:1]: This might generate some issues with identically named groups and compilers. Can you please prepend some common prefix to all of them please? Maybe `mirllc`, `mirllc33`,... And so on :)

File: etc/config/llvm_mir.defaults.properties
Comment (RubenRBS) [inline etc/config/llvm_mir.defaults.properties:5]: Same here, call it `mirllcdef` and so on please

**Review by RubenRBS — APPROVED** [review]

**Comment by RubenRBS** [comment]
Thanks a lot! This will be super useful!

**Review by matsjla — COMMENTED** [review]
Looks excellent, agree with the changes requested by @RubenRBS.

Maybe I'm missing something, but there's not a compiler.ts implementation here?

**Comment by yashssh** [comment]
> Maybe I'm missing something, but there's not a compiler.ts implementation here?

MIR relies on LLVM backend compiler (LLC) which was already added as one of the compilers for LLVM IR, hence separate implementation is not needed.

File: etc/config/llvm_mir.amazon.properties
Comment (RubenRBS) [inline etc/config/llvm_mir.amazon.properties:12]: If this uses `llc`, I think you'll need to set `.compilerType=llc` for the group as @matsjla figured out

Comment (yashssh) [reply etc/config/llvm_mir.amazon.properties:12]: I didn't get you. What change should I make in ```group.mirllc.compilerType=llc``` ?

Comment (RubenRBS) [reply etc/config/llvm_mir.amazon.properties:12]: Oh, nothing. Github app bugged and I didn't actually see that this line was as already should be! So all good to go :)

**Review by RubenRBS — APPROVED** [review]

**Comment by jeremy-rifkin** [comment]
Happy to merge once the merge conflict is resolved! (or if you'd like we can also resolve it 😄)

**Comment by yashssh** [comment]
> Happy to merge once the merge conflict is resolved! (or if you'd like we can also resolve it 😄)

Resolved it :)

**Comment by jeremy-rifkin** [comment]
Sweet! Thanks!

**Pull request merged**
